src/NuGet.Services.Metadata.Catalog.Monitoring/Monitoring/DatabasePackageStatusOutdatedCheckSource.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Services.Metadata.Catalog.Helpers;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Monitoring
+{
+    /// <summary>
+    /// Fetches <see cref="PackageStatusOutdatedCheck"/> from <see cref="IGalleryDatabaseQueryService.GetPackagesEditedSince(DateTime, int)"/>.
+    /// The <see cref="PackageStatusOutdatedCheck"/>s fetched represent the latest state for existing packages.
+    /// </summary>
+    public class DatabasePackageStatusOutdatedCheckSource : PackageStatusOutdatedCheckSource<FeedPackageDetails>
+    {
+        private readonly IGalleryDatabaseQueryService _galleryDatabaseQueryService;
+
+        public DatabasePackageStatusOutdatedCheckSource(
+            ReadWriteCursor cursor,
+            IGalleryDatabaseQueryService galleryDatabaseQueryService)
```

Comment:
Should we call the argument `galleryDatabase` for consistency with other places where we inject `IGalleryDatabaseQueryService`?

Reply:
Yup, my bad--keep forgetting what we call it!.